dask/array/routines.py

```diff
@@ -1843,6 +1843,22 @@ def piecewise(x, condlist, funclist, *args, **kw):
     )
 
 
+@derived_from(np)
+def select(condlist, choicelist, default=0):
+    return blockwise(
+        np.select,
+        tuple(range(1, choicelist[0].ndim + 1)),
+        condlist,
+        tuple(range(choicelist[0].ndim + 1)),
+        choicelist,
+        tuple(range(choicelist[0].ndim + 1)),
+        dtype=result_type(*choicelist),
+        name="select",
+        concatenate=True,
```

Comment:
One thing we were discussing in the issue is this may lead to performance issues. We might not solve this in this PR, but wanted to raise that concern so that if we see a way to fix it here we can do that or we can track it for follow up

Reply:
The high memory footprint related to `concatenate=true` could be the cause of the failing tests in CI? cc @jsignell

Reply:
It's certainly possible. This would cause higher memory usage